dot/state/block.go

```diff
@@ -250,14 +250,62 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetAllDescendants gets all the descendants even if hash is not stored in memory.
+func (bs *BlockState) GetAllDescendants(hash common.Hash) ([]common.Hash, error) {
+	allDescendants, err := bs.bt.GetAllDescendants(hash)
```

Comment:
nit have named returns? It always helps at some point documentation-wise

```suggestion
func (bs *BlockState) GetAllDescendants(hash common.Hash) (allDescendants []common.Hash, err error) {
	allDescendants, err = bs.bt.GetAllDescendants(hash)
```

Reply:
LOL